pallets/utility/src/lib.rs

```diff
@@ -148,5 +175,45 @@ decl_module! {
 
             Ok(())
         }
+
+        /// TODO docs
+        #[weight = SimpleDispatchInfo::FixedNormal(10_000)] // TODO replace this placeholder
+        pub fn relay_tx(
+            origin,
+            target: T::AccountId,
+            signature: <T as IdentityModule>::OffChainSignature,
+            call: UniqueCall<<T as Trait>::Call>
+        ) -> DispatchResult {
+            let origin = ensure_signed(origin)?;
+
+           let origin_key = AccountKey::try_from(origin.encode())?;
+           let target_key = AccountKey::try_from(target.encode())?;
+
+           let target_nonce = <Nonces<T>>::get(&target);
+
+            ensure!(
+                target_nonce == call.nonce,
+                Error::<T>::InvalidNonce
+            );
+
+            ensure!(
+                signature.verify(call.encode().as_slice(), &target),
+                Error::<T>::InvalidSignature
+            );
+
+            ensure!(
+                T::CddChecker::check_key_cdd(&origin_key),
+                Error::<T>::OriginCddMissing
+            );
+
+            ensure!(
+                T::CddChecker::check_key_cdd(&target_key),
+                Error::<T>::TargetCddMissing
+            );
+
+            <Nonces<T>>::insert(target.clone(), target_nonce + 1);
+
+            call.inner.dispatch(RawOrigin::Signed(target).into())
```

Comment:
Capture the result of this dispatch and return it as the result of this function.
something like
```
call.inner.dispatch(RawOrigin::Signed(target).into()).map(|_| ()).map_err(|e| e.error)
```

Reply:
I thought it was already doing that because `fn dispatch(self, origin: Self::Origin) -> crate::DispatchResult` and there is no `;` so that should return DispatchResult from the dispatch call.

Reply:
Seems like substrate finally implemented the `From`/`Into` trait for it. We can skip this change for now and visit it later if substrate is doing something weird in their conversion.